src/Raven.Server/Documents/Queries/Results/QueryResultRetrieverBase.cs

```diff
@@ -255,9 +375,24 @@ protected void FinishDocumentSetup(Document doc, Lucene.Net.Search.ScoreDoc scor
                             if (FieldsToFetch.SingleBodyOrMethodWithNoAlias)
                             {
                                 if (fieldVal is BlittableJsonReaderObject nested)
-                                    doc.Data = nested;
+                                {
+                                    if (ReferenceEquals(nested, doc.Data) == false)
+                                    {
+                                        using (var _ = doc)
```

Comment:
You can have `using (doc)`

Reply:
I've changed it into
```csharp
using (_ = doc)
   //operation
```

since pattern 
```csharp
using (doc)
    doc = operation
```

gets a warning and I wanted to avoid a lot of pragma disable.